xbmc/guilib/GUIAction.h

```diff
@@ -16,26 +16,74 @@ class CGUIControl;
 class CGUIListItem; typedef std::shared_ptr<CGUIListItem> CGUIListItemPtr;
 
 /**
- * Class containing vector of condition->(action/navigation route) pairs and handling its execution.
+ * Class containing vector of condition->(action/navigation route) and handling its execution.
  */
 class CGUIAction
 {
 public:
+  /**
+   * Structure containing the the executable action definition
+   */
+  struct ExecutableAction
+  {
+  public:
+    /**
+    * Executable action default constructor
+    */
+    ExecutableAction() = default;
```

Comment:
Why? Does the instance not work in case somebody prefers to use default ctor and dedicated setters? `delete`should only be used if the instance does not work when constructed this way.

Reply:
> although struct and class are equal

What? Not really, see https://stackoverflow.com/questions/92859/what-are-the-differences-between-struct-and-class-in-c

Reply:
>  although struct and class are equal

What? Not really, see https://stackoverflow.com/questions/92859/what-are-the-differences-between-struct-and-class-in-c